Refactor localization

mlbiche opened the pull request:
- Fix localization asked on each pages (Fix #324 )
- Clarify localization code by centralising its code

EDIT :
- Fix Around map not focused on user context (Fix #349 )

mlbiche committed:
Refactor localization

- Fix localization asked on each pages
- Clarify localization code by centralising its code

fuentesloic commented:
Just a question relative to flask I guess. Does it make sens to have big template because we set all the `js` on it? is not better to split html/js?

mlbiche committed:
Fix wrong naming

mlbiche commented:
@fuentesloic Not sure about what you meant through your question. 🤔 Do you have an example ? Are you talking about the localize.html file ? I am pretty new to Flask and Jinja so, I am not trying to change the existing architecture, just to refactor some pieces of code.

fuentesloic commented:
@vinyll up?

vinyll commented on static/js/utils.js:
```diff
@@ -21,25 +21,4 @@ function notify (type, message, delay) {
-    navigator.geolocation.getCurrentPosition(success, fail, options)
-  })
-}
+}
```
Missing an newline

vinyll commented on templates/script/localize.html:
```diff
@@ -1,13 +1,31 @@
+            window.location.reload()
         })
-      })
+      }
```
I see this code was moved from utils.js to an includable HTML. It is solving something?
The benefit with this code is utils.js is that it it cached by the browser for all HTML pages.
the _templates/script/localize.html_ is not being cached as it is get included _as is_ into each HTML page.

mlbiche committed:
Move localisation related code to JS file

vinyll commented on static/js/utils.js:
```diff
@@ -1,10 +1,38 @@
+    moveTo(localizeUserPosition)
+  if (window.location.pathname === "{{ url_for('form_spot') }}")
+    userPosition = localizeUserPosition
+}
```
I guess this means that we want a specific action once the usser is localized.
It could be translated with something like:
```
if(userLocalized) userLocalized(localizeUserPosition)
```

Where `userLocalized` would be a function that is specific within the templates "around" and "form_spot"

vinyll commented on templates/around/form_spot.html:
```diff
@@ -51,9 +51,7 @@ <h2>Share your place</h2>
-    let userLocation = {{ current_user.current_location }}
-    // try to get browser position from promise in utils if the user do not share its position
-    if (!userLocation) userLocation = getPosition()
+    let userPosition // NOTE : Its value is affected from localize.html script
```
Not sure if this is actually useful; `let` is local scoping.

vinyll commented on templates/script/location-trips.html:
```diff
@@ -1,5 +1,4 @@
 <script>
-	var current_location = {{ user.current_location or [] }}
 	var locations = [
```
Not sure why we have 4 spaces in this file while it's 2 spaces all around.
I not this is out of this PR, just noticing.

mlbiche committed:
Update use of current location

- Update around map to move to user location once he has allowed geolocation. Fix #349
- Remove dead code

Note that if the user directly navigates to Trips before allowing geolocation, his current location won't appear once he allows it. It will only appear on refresh.

mlbiche commented:
Also fix #349

vinyll commented on static/js/map-around.js:
```diff
@@ -1,6 +1,7 @@
 // POIS creation
 const worker = new Worker('/static/js/webworker-around.js')
 let pois = []
+let currentUserLocation
```
Why do we need this here as `currentUserLocation` is only used in the `userLocalized` function?

mlbiche committed:
Remove useless global variable for current location

vinyll reviewed (approved):
:)
